docs/permission/writing-a-policy.md

```diff
@@ -3,3 +3,137 @@ id: writing-a-policy
+
+The `isEntityOwner` rule needs a list of claims as a parameter, and the second argument to our `handle` method provides us with a `BackstageIdentityResponse` object, from which we can grab the user's `ownershipEntityRefs`. We provide an empty array as a fallback since the user may be anonymous.
+
+You should now be able to see in your Backstage app that the unregister entity button is enabled for entities that you own, but disabled for all other entities!
+
+## Resource types
+
+Now let's say we would also like to restrict users from viewing catalog entities that they do not own, just like we did for unregistering entities. One way to achieve this may be to simply duplicate our if statement and check for the `catalog.entti.read` permission:
+
+```diff
+    async handle(
+      request: PolicyAuthorizeQuery,
+      user?: BackstageIdentityResponse,
+     ): Promise<PolicyDecision> {
+      if (request.permission.name === 'catalog.entity.delete') {
+        return createCatalogPolicyDecision(
+          catalogConditions.isEntityOwner(
+            user?.identity.ownershipEntityRefs ?? [],
+          ),
+        );
+      }
+
++     if (request.permission.name === 'catalog.entity.read') {
++       return createCatalogPolicyDecision(
++         catalogConditions.isEntityOwner(
++           user?.identity.ownershipEntityRefs ?? [],
++         ),
++       );
++     }
+
+      return { result: AuthorizeResult.ALLOW };
+    }
+```
+
+If you choose to write your policy this way, it will certainly work! You should be able to verify this by saving this policy and seeing that the catalog now only shows the entities that you own. However, you can imagine that as policies grow to handle many different permissions, these conditionals can quickly become repetitive. We can author this same policy in a more scalable way by using resource types.
+
+```diff
+    async handle(
+      request: PolicyAuthorizeQuery,
+      user?: BackstageIdentityResponse,
+     ): Promise<PolicyDecision> {
+-     if (request.permission.name === 'catalog.entity.delete') {
++     if (request.permission.resourceType === 'catalog-entity') {
+        return createCatalogPolicyDecision(
+          catalogConditions.isEntityOwner(
+            user?.identity.ownershipEntityRefs ?? [],
+          ),
+        );
+      }
+
+-     if (request.permission.name === 'catalog.entity.read') {
+-       return createCatalogPolicyDecision(
+-         catalogConditions.isEntityOwner(
+-           user?.identity.ownershipEntityRefs ?? [],
+-         ),
+-       );
+-     }
+-
+      return { result: AuthorizeResult.ALLOW };
+    }
+```
+
+In this example, we use the `catalog-entity` resource type to catch all authorization requests that have to do with resources from the catalog. Now, you should be able to see the same functionality as before (only see the catalog entities that you own) - success!
```

Comment:
I think this is worded really nicely, but it feels like there's a possibility for miscommunication - perhaps we should add a note at the bottom pointing out the fact that `catalog.entity.create` doesn't have the `catalog-entity` resource-type, because you can't apply conditions to entities that don't yet exist.

Reply:
Good callout, will add a note.